### PR TITLE
[ENT-1616] Remove branding property from pool and all direct references of this property 

### PR DESCRIPTION
**Pull request by wolfdale**
- Branched out from `nikos/f/pooldto_branding_from_product`

### CODE REVIEW

File: server/src/main/java/org/candlepin/model/Pool.java [hunk 1]
```diff
@@ -82,7 +81,7 @@
 @Table(name = Pool.DB_TABLE)
 @JsonFilter("PoolFilter")
 public class Pool extends AbstractHibernateObject<Pool> implements Owned, Named, Comparable<Pool>,
-    Eventful, SubscriptionInfo {
```
[the remark this answers is not in the document]
Comment (Ceiu) [reply server/src/main/java/org/candlepin/model/Pool.java]: The ```*Info``` interfaces are the bridge between the external subscription model (IT/adapters) and our model, which directly impacts manifest import and refresh. For the immediate term, we'll need to keep that as-is, even if the ```getBranding()``` method just returns null.

This whole thing is an area that, at some point, we'll need to work with IT to figure out how to resolve the model change, and/or we'll need to have some kludge code in place to let their model work with ours.

[the remark this answers is not in the document]
Comment (nikosmoum) [reply server/src/main/java/org/candlepin/model/Pool.java]: Ok, so we add the interface back, and the getBranding() method to return null.

**Comment by nikosmoum** [comment]
A note: This PR should be merged only once ENT-1612 & ENT-1617 are merged, and after the target branch is switched from nikos/f/pooldto_branding_from_product to feature/product_branding.

File: server/src/main/java/org/candlepin/model/ConsumerCurator.java [hunk 2]
```diff
@@ -982,7 +982,7 @@ public Consumer verifyAndLookupConsumerWithEntitlements(String consumerUuid) {
             Hibernate.initialize(e.getCertificates());
 
             if (e.getPool() != null) {
-                Hibernate.initialize(e.getPool().getBranding());
+                Hibernate.initialize(e.getPool().getProduct().getBranding());
```
[the remark this answers is not in the document]
Comment (Ceiu) [reply server/src/main/java/org/candlepin/model/ConsumerCurator.java]: This sounds about right

File: server/src/main/java/org/candlepin/model/Pool.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ceiu) [reply server/src/main/java/org/candlepin/model/Pool.java]: We could also throw an UnsupportedOperationException if we really want to emphasize that it shouldn't be used.

File: server/src/main/java/org/candlepin/model/dto/Subscription.java [hunk 3]
```diff
@@ -608,9 +609,19 @@ public Subscription populate(Pool source, ProductCurator productCurator) {
         this.setUpstreamEntitlementId(source.getUpstreamEntitlementId());
         this.setUpstreamConsumerId(source.getUpstreamConsumerId());
 
-        //TODO: This should get updated to populate the branding from the pool's main product once Pool
-        // stops having a reference to it. (ENT-1616)
-        this.setBranding(source.getBranding());
+        Collection<Branding> brand = source.getProduct().getBranding();
```
[the remark this answers is not in the document]
Comment (Ceiu) [reply server/src/main/java/org/candlepin/model/dto/Subscription.java]: :+1:

[the remark this answers is not in the document]
Comment (Ceiu) [reply server/src/main/java/org/candlepin/model/dto/Subscription.java]: :+1:

File: server/src/main/java/org/candlepin/model/ConsumerCurator.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Ceiu) [reply server/src/main/java/org/candlepin/model/ConsumerCurator.java]: This sounds about right

File: server/src/main/java/org/candlepin/model/Branding.java [hunk 4]
```diff
@@ -67,14 +73,19 @@
...
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "product_uuid")
+    private Product product;
+
     public Branding() {
         // Intentionally left empty
     }
 
-    public Branding(String productId, String type, String name) {
+    public Branding(String productId, String type, String name, Product parent) {
```
[the remark this answers is not in the document]
Comment (wolfdale) [reply server/src/main/java/org/candlepin/model/Branding.java]: Alright.
@nikosmoum  Will it affect other PR as well, as its a common constructor. or Should I go ahead and change it in this PR.

[the remark this answers is not in the document]
Comment (nikosmoum) [reply server/src/main/java/org/candlepin/model/Branding.java]: @wolfdale Feel free to change it in your PR

File: server/src/main/java/org/candlepin/model/Branding.java [hunk 5]
```diff
@@ -138,20 +166,31 @@ public boolean equals(Object anObject) {
...
+            .append(this.productId)
+            .append(this.type)
+            .append(this.getProduct() != null ? this.getProduct().getUuid() : null)
+            .toHashCode();
     }
 
     @Override
     public String toString() {
-        return String.format("Branding [id: %s, name: %s, productId: %s, type: %s]",
+        return String.format("ProductBranding [id: %s, name: %s, productId: %s, type: %s]",
```
[the remark this answers is not in the document]
Comment (wolfdale) [reply server/src/main/java/org/candlepin/model/Branding.java]: Corrected.

File: server/src/main/java/org/candlepin/model/ConsumerCurator.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (wolfdale) [reply server/src/main/java/org/candlepin/model/ConsumerCurator.java]: sure, removing it.

File: server/src/main/java/org/candlepin/model/Pool.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (wolfdale) [reply server/src/main/java/org/candlepin/model/Pool.java]: Added the interface back, and the getBranding() method to return null.

File: server/src/main/java/org/candlepin/model/dto/Subscription.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (wolfdale) [reply server/src/main/java/org/candlepin/model/dto/Subscription.java]: Fixed.

File: server/src/test/java/org/candlepin/dto/manifest/v1/PoolTranslatorTest.java [hunk 6]
```diff
@@ -89,7 +88,8 @@ protected Pool initSourceObject() {
 
         Set<Branding> brandingSet = new HashSet<>();
         brandingSet.add(this.brandingTranslatorTest.initSourceObject());
-        source.setBranding(brandingSet);
+        Product product = source.getProduct();
+        product.setBranding(brandingSet);
```
[the remark this answers is not in the document]
Comment (wolfdale) [reply server/src/test/java/org/candlepin/dto/manifest/v1/PoolTranslatorTest.java]: Removed. I overlooked that Product object is mocked.

File: server/src/test/java/org/candlepin/test/TestUtil.java [hunk 7]
```diff
@@ -679,10 +678,18 @@ public static Pool clone(Pool pool) {
         // Copy sub-product data if there is any:
         p.setDerivedProduct(pool.getDerivedProduct());
 
-        for (Branding b : pool.getBranding()) {
-            p.getBranding().add(new Branding(b.getProductId(), b.getType(), b.getName()));
+        //Copying Branding
```
[the remark this answers is not in the document]
Comment (wolfdale) [reply server/src/test/java/org/candlepin/test/TestUtil.java]: Alright, removing it.

File: server/src/test/java/org/candlepin/test/TestUtil.java [hunk 8]
```diff
@@ -771,7 +778,7 @@ public static void assertPoolsAreEqual(Pool pool1, Pool pool2) {
         assertEquals(pool1.getEntitlements(), pool2.getEntitlements());
         assertEquals(pool1.getConsumed(), pool2.getConsumed());
         assertEquals(pool1.getExported(), pool2.getExported());
-        assertEquals(pool1.getBranding(), pool2.getBranding());
+        assertEquals(pool1.getProduct().getBranding(), pool2.getProduct().getBranding());
```
[the remark this answers is not in the document]
Comment (wolfdale) [reply server/src/test/java/org/candlepin/test/TestUtil.java]: Removed.

File: server/src/test/java/org/candlepin/util/X509V3ExtensionUtilTest.java [hunk 9]
```diff
@@ -154,7 +153,9 @@ public void productWithBrandName() {
         Set<Product> prods = new HashSet<>(Arrays.asList(p));
         Product mktProd = new Product("mkt", "MKT SKU");
         Pool pool = TestUtil.createPool(mktProd);
-        pool.getBranding().add(new Branding(engProdId, "OS", brandedName));
+        Product product = pool.getProduct();
+        product.setBranding(Arrays.asList(new Branding(engProdId, "OS", brandedName, null)));
+        pool.setProduct(product);
```
[the remark this answers is not in the document]
Comment (wolfdale) [reply server/src/test/java/org/candlepin/util/X509V3ExtensionUtilTest.java]: Corrected.

File: server/src/test/java/org/candlepin/util/X509V3ExtensionUtilTest.java [hunk 10]
```diff
@@ -176,11 +177,15 @@ public void productWithMultipleBrandNames() {
...
-        pool.getBranding().add(new Branding(engProdId, "OS", brandedName));
-        pool.getBranding().add(new Branding(engProdId, "OS", "another brand name"));
-        pool.getBranding().add(new Branding(engProdId, "OS", "number 3"));
+        Product product = pool.getProduct();
+        List<Branding> branding = new LinkedList<>();
+        branding.add(new Branding(engProdId, "OS", brandedName, null));
+        branding.add(new Branding(engProdId, "OS", "another brand name", null));
+        branding.add(new Branding(engProdId, "OS", "number 3", null));
+        product.setBranding(branding);
+        pool.setProduct(product);
```
[the remark this answers is not in the document]
Comment (wolfdale) [reply server/src/test/java/org/candlepin/util/X509V3ExtensionUtilTest.java]: Corrected.

File: server/src/main/java/org/candlepin/model/Branding.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (wolfdale) [reply server/src/main/java/org/candlepin/model/Branding.java]: Changed the constructor signature.

File: server/src/main/java/org/candlepin/resource/OwnerResource.java [hunk 11]
```diff
@@ -791,23 +789,6 @@ protected void populateEntity(Pool entity, PoolDTO dto) {
             }
         }
 
-        if (dto.getBranding() != null) {
```
[the remark this answers is not in the document]
Comment (wolfdale) [reply server/src/main/java/org/candlepin/resource/OwnerResource.java]: `@JsonIgnore` added only for `org.candlepin.dto.api.v1.PoolDTO` class.

**Comment by nikosmoum** [comment]
@wolfdale Unfortunately, I've had to rework my original refresh changes to update the Adapters API, which was a lot of changes, so this PR has to be rebased & conflicts resolved :/

**Comment by wolfdale** [comment]
@nikosmoum sure, that's alright. Will take care of it.

**Pull request closed without merging**